pandas/tests/frame/test_constructors.py

```diff
@@ -259,6 +259,12 @@ def test_constructor_dict(self):
         frame = DataFrame({'A': [], 'B': []}, columns=['A', 'B'])
         self.assert_index_equal(frame.index, Index([], dtype=np.int64))
 
+        # GH 14381
+        # Dict with None value
+        frame_none = DataFrame(dict(a=None), index=[0])
+        frame_nan = DataFrame(dict(a=[None]), index=[0])
+        tm.assert_frame_equal(frame_none, frame_nan)
```

Comment:
Can you add an assert to check that the value in the frame is actually `None` (and not np.nan)